tests/TestCase/ORM/Association/HasOneTest.php

```diff
@@ -79,6 +79,21 @@ public function tearDown()
     }
 
     /**
+     * Tests that foreignKey() returns the correct configured value
+     *
+     * @return void
+     */
+    public function testForeignKey()
+    {
+        $assoc = new HasOne('Profiles', [
+            'sourceTable' => $this->user
+        ]);
+        $this->assertEquals('user_id', $assoc->foreignKey());
+        $this->assertEquals('another_key', $assoc->foreignKey('another_key'));
+        $this->assertEquals('another_key', $assoc->foreignKey());
```

Comment:
If they are necessary, I can add tests for new setters/getters of Association classes. Adding all new setters/getters are hard for me 😭